### ISSUE TITLE
Cannot save tables content

### ISSUE THREAD

**Issue by amadeous** [mirko-pagliai/cakephp-database-backup#25]
I got the setup to back up a database but the content is not saved. I searched for configuration of tables to backup but found nothing.
Could you help me ?

**Comment by mirko-pagliai** [comment]
Hello. Please provide all the information that may be useful: PHP version,
CakePHP version, database type, what database contains, how are you
exporting

Il Dom 1 Apr 2018, 22:13 Amadeous <notifications@github.com> ha scritto:

**Comment by mirko-pagliai** [comment]
it seems that it behaves as if there were no database. Is the connection to the database the `default` one? Can you also try to export using the shell?

**Comment by amadeous** [comment]
I found the problem. The error was due to permission access to the database. 

When testing with the shell, I was surprised to notice that when I tried to export without any option, I got the (not perfectly clear) message in red :
```
Failed with exit code '2'
```
but when I tried with the option `-c gzip`, I got a much clearer message but not in red
```
mysqldump: Got error: 1044: Access denied for user 'mydbuser'@'%' to database 'myDatabase' when using LOCK TABLES
```

The user I created for the application only had `SELECT`, `INSERT`, `UPDATE`, `DELETE`, `EXECUTE` and `SHOW VIEW`. I added the `LOCK TABLES` right to my user and it worked !

Thank you for your help and your great plugin !
